packages/apify/src/crawlers/cheerio_crawler.ts

```diff
@@ -651,7 +652,7 @@ export class CheerioCrawler extends BasicCrawler {
         });
 
         return addTimeoutToPromise(
-            this.userProvidedHandler(crawlingContext),
+            Promise.resolve().then(() => this.userProvidedHandler(crawlingContext)),
```

Comment:
isn't this the same as `Promise.resolve(this.userProvidedHandler(crawlingContext))`?

maybe we should just allow `Awaitable` in the `addTimeoutToPromise` helper so we don't have to deal with this?

Reply:
Oh, we can do `Promise.resolve()` instead, yeah. 

I was thinking that too, and was going to do it, but said helper validates that it receives a promise.. I can apply the refactor to `Promise.resolve()`, thoughts?

Reply:
so it would actually fail on validation if not given async function? how come the tests were passing before with the sync functions used?

Reply:
I.. would love to have an answer to that.. I know the tests compiled before because the functions were mandatory to return Promises so I made all tests use async functions... I'll do what you suggest would be the best solution